Description: use VA_COPY() in Shell
 Upstream edk2 has switched to using ms abi intrinsic va_list handling
 for x86-64.  Fix the Shell to do proper va_list handling instead of trying
 to treat them like pointers.
Author: Steve Langasek <vorlon@debian.org>

Index: edk2/Shell/Library/IO.c
===================================================================
--- edk2.orig/Shell/Library/IO.c
+++ edk2/Shell/Library/IO.c
@@ -283,7 +283,7 @@
   ps.Output   = Output;
   ps.Context  = spc;
   ps.fmt.u.pw = fmt;
-  ps.args     = args;
+  VA_COPY(ps.args, args);
   _Print (&ps);
 }
 
@@ -635,7 +635,7 @@
     ps.fmt.u.pc   = fmta;
   }
 
-  ps.args = args;
+  VA_COPY(ps.args, args);
 
   if (Column != (UINTN) -1) {
     Out->SetCursorPosition (Out, Column, Row);
@@ -1984,7 +1984,7 @@
   ps.Output     = _DbgOut;
   ps.fmt.Ascii  = TRUE;
   ps.fmt.u.pc   = fmt;
-  ps.args       = args;
+  VA_COPY(ps.args, args);
   ps.Attr       = EFI_TEXT_ATTR (EFI_LIGHTGRAY, EFI_RED);
 
   DbgOut        = LibRuntimeDebugOut;
